tests/integration/test_agentd/test_agentd_server_address_configuration.py

```diff
@@ -0,0 +1,97 @@
+local_internal_options = {'agent.debug': '2'}
+
+parameters = [
+    {'SERVER_ADDRESS': 'MANAGER_IP'},                               # Invalid server address
+    {'SERVER_ADDRESS': '172.28.128.hello'},                         # Could not resolve hostname
+    {'SERVER_ADDRESS': '172.28.128.12'},                            # Valid IP, unable to connect (IPv4)
+    {'SERVER_ADDRESS': '::ffff:ac1c:800c'},                         # Valid IP, unable to connect (IPv6 compressed)
+    {'SERVER_ADDRESS': '0000:0000:0000:0000:0000:ffff:ac1c:800c'},  # Valid IP, unable to connect (IPv6 expanded)
+    {'SERVER_ADDRESS': 'unable'},                                   # Resolve hostname, valid IP, unable to connect (IPv4)
+    {'SERVER_ADDRESS': 'unable6compressed'},                        # Resolve hostname, valid IP, unable to connect (IPv6 compressed)
+    {'SERVER_ADDRESS': 'unable6'}                                   # Resolve hostname, valid IP, unable to connect (IPv6 expanded)
+]
+
+metadata = [
+    {'server_address': 'MANAGER_IP',                              'invalid_address': True,  'resolve_hostname': False},
+    {'server_address': '172.28.128.hello',                        'invalid_address': False, 'resolve_hostname': False},
+    {'server_address': '172.28.128.12',                           'invalid_address': False, 'resolve_hostname': True},
+    {'server_address': '::ffff:ac1c:800c',                        'invalid_address': False, 'resolve_hostname': True},
+    {'server_address': '0000:0000:0000:0000:0000:ffff:ac1c:800c', 'invalid_address': False, 'resolve_hostname': True},
+    {'server_address': 'unable', 'invalid_address': False, 'resolve_hostname': True, 'real_ip': '172.28.128.12'},
+    {'server_address': 'unable6compressed', 'invalid_address': False, 'resolve_hostname': True, 'real_ip': '::ffff:ac1c:800c'},
+    {'server_address': 'unable6', 'invalid_address': False, 'resolve_hostname': True, 'real_ip': '0000:0000:0000:0000:0000:ffff:ac1c:800c'}
+]
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f"{x['SERVER_IP']}" for x in parameters]
+
+# Functions
+
+
+# Fixtures
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+@pytest.fixture(scope="module")
+def edit_hosts(get_configuration):
+    if 'real_ip' in get_configuration['metadata']:
+        with open('/etc/hosts', 'r+') as file:
+            original_content = file.read()
+
+            new_content = get_configuration['metadata']['real_ip'] + ' ' + get_configuration['metadata']['server_address']
+            file.write(new_content)
+
+    yield
+
+    if 'real_ip' in get_configuration['metadata']:
+        with open('/etc/hosts', 'w') as file:
+            file.write(original_content)
```

Comment:
This will fail for Windows systems. It is needed to change this path for Windows agents. Try  this
`C:\Windows\System32\Drivers\etc\hosts`. Also, it will be recommended to create a tool variable with this path, because maybe this will be used in the future.